### PR TITLE
Implement extended hbl configuration for AMS 0.10.2

### PR DESCRIPTION
**Pull request by cathery**
- Override key settings now support up to 8 custom title overrides.
  - Album override can now also be changed.
  - For this, a title list selection was added. It uses a separate thread with turbojpeg to asynchronously load and display title info to avoid stutter.
- All buttons now autoscroll when the selection goes out of bounds.
  - This means the sysmodules page no longer needs pages. They were removed.
- Fixed an issue where program_id_0 and override_key_0 was not properly read from the config.
- Fixed touchscreen UI not working properly when inside a list selector and the message box.
- Fixed Override key button not blocking input properly when reading for button presses.
- Added a description to Application override settings.
- Added an option to immediately exit the program by pressing PLUS.
- Remember last selected option in most menus.
- Add a warning about applet memory pool when an applet is selected in key override menu.
- Lots of minor UI changes

### CODE REVIEW

File: include/gui/gui_overrides_menu.hpp [hunk 1]
```diff
@@ -4,25 +4,11 @@
 #include "override_key.hpp"
 
 enum class OverrideButtonType {
-  Album,
   Any_Title,
   Custom_Title,
   AddNew,
```
Comment (WerWolv) [inline include/gui/gui_overrides_menu.hpp]: Should probably be Add_New or make everything UPPER_SNAKE_CASE like all other enums

File: include/utils/override_key.hpp [hunk 2]
```diff
@@ -2,8 +2,12 @@
 #include <string>
 #include <switch.h>
 
+#define OVERRIDE_KEY "override_key_0"
+#define PROGRAM_ID "program_id_0"
+
 enum class OverrideKeyType {
-  Override0 = 0,  //default when un-numbered
+  Invalid = -1,
```
Comment (WerWolv) [inline include/utils/override_key.hpp:9]: UPPER_SNAKE_CASE as well?

File: include/gui/button.hpp [hunk 3]
```diff
@@ -10,6 +10,15 @@
 class Button {
 public:
   static inline std::vector<Button*> g_buttons;
```
Comment (WerWolv) [inline include/gui/button.hpp:12]: Prefix public static inline variables with `g_`

**Review by WerWolv — CHANGES_REQUESTED** [review]

File: include/gui/button.hpp (hunk 3 above)
Comment (cathery) [reply include/gui/button.hpp:12]: Resolved in 44e6aaf

File: include/gui/gui_overrides_menu.hpp (hunk 1 above)
Comment (cathery) [reply include/gui/gui_overrides_menu.hpp]: Resolved in 531c030

**Review by WerWolv — APPROVED** [review]

**Review by WerWolv — APPROVED** [review]

**Pull request merged**